pkg/cni/cilium/cilium_test.go

```diff
@@ -144,6 +144,32 @@ func TestValidateCiliumValuesUpdate(t *testing.T) {
 			},
 			expectedError: "[spec.values.operator.securityContext: Invalid value: \"null\": value is immutable]",
 		},
+		{
+			name: "cni excluded field introduced under immutable value",
+			testValuesModifier: func(values map[string]any) {
+				o := values["cni"].(map[string]any)
+				o["chainingMode"] = "portmap"
+			},
+			expectedError: "[]",
+		},
+		{
+			name: "ipam excluded field introduced under immutable value",
+			testValuesModifier: func(values map[string]any) {
+				ipam := values["ipam"].(map[string]any)
+				op := ipam["operator"].(map[string]any)
+				op["clusterPoolIPv4PodCIDR"] = "192.168.0.0/24"
+			},
+			expectedError: "[]",
+		},
+		{
+			name: "Modified immutable nested value in ipam",
+			testValuesModifier: func(values map[string]any) {
+				ipam := values["ipam"].(map[string]any)
+				op := ipam["operator"].(map[string]any)
+				op["clusterPoolIPv4PodCIDRList"] = []string{"192.168.0.0/24"}
+			},
+			expectedError: "[spec.values.ipam: Invalid value: map[string]interface {}{\"operator\":map[string]interface {}{\"clusterPoolIPv4MaskSize\":\"16\", \"clusterPoolIPv4PodCIDRList\":[]string{\"192.168.0.0/24\"}}}: value is immutable]",
+		},
```

Comment:
I don't understand this test. Don't we want to specifically _allow_ this? Why would we expect an error here?

Reply:
We only want to allow `clusterPoolIPv4PodCIDR` to unblock upgrade
This checks if the changes only to `clusterPoolIPv4PodCIDRList` are still blocked.

Reply:
Okay, gotcha.